### PR TITLE
Restore and tweak the usage of crm_mon as CGI script + crm_ends_with special case optimization

### PR DESCRIPTION
**Pull request by jnpkrn**
Replacement for PR #1299.
For the possible integration with httpd, see commit message of the 5th commit.

### CODE REVIEW

File: tools/crm_mon.c [hunk 1]
```diff
@@ -555,7 +556,7 @@ main(int argc, char **argv)
     signal(SIGCLD, SIG_IGN);
 #endif
 
-    if (strcmp(crm_system_name, "crm_mon.cgi") == 0) {
+    if (safe_str_eq(strrchr(argv[0], '.'), ".cgi") == TRUE) {
```
Comment (kgaillot) [inline tools/crm_mon.c]: See crm_ends_with()

**Comment by jnpkrn** [comment]
Should I add a note to the help screen about the conflict/mutual exclusion
of the modes that now became intentionally enforced?

**Comment by jnpkrn** [comment]
Ditto for the actual `X.cgi` mode of launching...

File: tools/crm_mon.c [hunk 2]
```diff
@@ -696,13 +702,29 @@ main(int argc, char **argv)
         }
     }
 
-    if (optind < argc) {
+    /* Extra sanity check when in CGI mode */
+    if (output_format != mon_output_cgi) {
```
Comment (kgaillot) [inline tools/crm_mon.c]: ==?

Comment (jnpkrn) [inline tools/crm_mon.c]: Thanks for catching, burnt by reordering patches and somehow reintroducing the thinko, I've been there already :-/

**Comment by jnpkrn** [comment]
I've figured out that I had to redo the SELinux part a bit, was working
just with hardlinks and when accidentally touching the label of the
original.

File: tools/crm_mon.c (hunk 1 above)
Comment (jnpkrn) [inline tools/crm_mon.c]: Thanks, though original implementation wasn't too optimal :)

**Comment by jnpkrn** [comment]
(pengine tests borked again, incentive to accept _anything_ via PRs?)

**Comment by jnpkrn** [comment]
Just extended the commit message of the last commit with the mention of
pacemaker ACLs (for instance, SELinux itself is too coarse-grained here).

**Comment by jnpkrn** [comment]
Re pengine tests: indeed, I haven't rebased.  Will do.
Anyway, that would not have happened if there was always-green
rule, which currently it isn't (modulo whole approach being racy,
leading some teams to adapt merge bots running the tests
on ephemeral merge just-in-time prior to real merge, which is
the most appealing workflow, TBH).

**Comment by jnpkrn** [comment]
Heck, I know what's going on, please do not merge.

**Comment by jnpkrn** [comment]
I owe several explanations here:

1. my entusiasm about the special case made me blind that I'll screw
   general cases -- which is in fact the real cause for the CI failure
   (well, I was originally interested in the special case in isolation,
   then was pointed out to the general case that I approached with the
   optics of the special case -- but in the end, it was worthy exercise
   and all the known special cases will benefit from the change)

2. I believed that known-to-be-defective base from Travis CI perspective
   for my own changes is somehow sticky -- unless I rebase explicitly;
   this is simply not true, the virtual merge against the target upstream
   branch (fixed by that time) is what Travis CI sees for it's routine:

> We rely on the merge commit that GitHub transparently creates between
> the changes in the source branch and the upstream branch the pull
> request is sent against.

https://docs.travis-ci.com/user/pull-requests/#My-Pull-Request-isn%E2%80%99t-being-built

**Comment by jnpkrn** [comment]
Extended commit message for one commit, in-code doctexts for another.
CI failure is unrelated, it was OK previously:
https://travis-ci.org/ClusterLabs/pacemaker/builds/255035134

**Pull request merged**